includes/wikia/Wikia.php

```diff
@@ -1254,60 +1252,6 @@ static public function parseMessageToArray($msgName, $forContent = false) {
 		wfProfileOut( __METHOD__ );
 		return $items;
 	}
-	/**
-	 * check user authentication key
-	 * @static
-	 * @access public
-	 * @param array $params
-	 */
-	static public function verifyUserSecretKey( $url, $hash_algorithm = 'sha256' ) {
-		global $wgWikiaAuthTokenKeys;
-		wfProfileIn( __METHOD__ );
-
-		@list( $user, $signature1, $signature2, $public_key ) = explode("|", base64_decode( strtr($url, '-_,', '+/=') ));
-
-		if ( empty( $user ) || empty( $signature1 ) || empty( $signature2 ) || empty ( $public_key) ) {
-			wfProfileOut( __METHOD__ );
-			return false;
-		}
-
-		# verification public key
-		if ( $wgWikiaAuthTokenKeys['public'] == $public_key ) {
-			$private_key = $wgWikiaAuthTokenKeys['private'];
-		} else {
-			wfProfileOut( __METHOD__ );
-			return false;
-		}
-
-		$oUser = User::newFromName( $user );
-		if ( !is_object($oUser) ) {
-			wfProfileOut( __METHOD__ );
-			return false;
-		}
-
-		// verify params
-		$email = $oUser->getEmail();
-		$params = array(
-			'user'			=> (string) $user,
-			'signature1'	=> (string) $signature1,
-			'token'			=> (string) wfGenerateUnsubToken( $email, $signature1 )
-		);
-
-		// message to hash
-		$message = serialize( $params );
-
-		// computed signature
-		$hash = hash_hmac( $hash_algorithm, $message, $private_key );
-
-		// compare values
-		if ( $hash != $signature2 ) {
-			wfProfileOut( __METHOD__ );
-			return false;
-		}
-
-		wfProfileOut( __METHOD__ );
-		return $params;
-	}
 
 	static public function isUnsubscribed( $to, $body, $subject ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$body". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$subject". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)